### ISSUE TITLE
Dose it support ControlNet inpaint in SDXL？

### ISSUE THREAD

**Issue by Whoo-jl** [megvii-research/HiDiffusion#16]

**Comment by ShenZhang-Shin** [comment]
Yes, I'm working on it. It works fine. I will integrate it into the code soon.
Here is an example. I directly edit 2K image using controlnet inpainting + hidiffusion
input: 
![christ](https://github.com/megvii-research/HiDiffusion/assets/58697142/c524f192-686f-440f-8a3c-5172297cf9fc)

output:
![hidi_christ_27_seed_65_conditioning_scale_0 5_religious or historical artwork, Christian tradition style  Alien or Xenomorph sits on a rocky outcr](https://github.com/megvii-research/HiDiffusion/assets/58697142/7c69c2b3-e788-422c-9a09-ba6d6c6e1977)

**Comment by Whoo-jl** [comment]
> Yes, I'm working on it. It works fine. I will integrate it into the code soon.是的，我正在努力。它工作正常。我将很快将其集成到代码中。 Here is an example. I directly edit 2K image using controlnet inpainting + hidiffusion下面是一个示例。我使用 controlnet inpainting + hidiffusion 直接编辑 2K 图像 input: 输入： ![christ](https://private-user-images.githubusercontent.com/58697142/329128448-c524f192-686f-440f-8a3c-5172297cf9fc.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMzM4MTEsIm5iZiI6MTcxNTIzMzUxMSwicGF0aCI6Ii81ODY5NzE0Mi8zMjkxMjg0NDgtYzUyNGYxOTItNjg2Zi00NDBmLThhM2MtNTE3MjI5N2NmOWZjLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOVQwNTQ1MTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNmUwYWI1ODc2ZmY3NzkyMjZjZGE1M2MwMDEyMmU0MTc4NzUzZGQ4NThhNjNhZWY5ODc0ZTIwM2U4MmJhNGIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.o6jc2LFWY2i86FgTp3JZLJ9zpYhBiMIeZBc68bgBQUY)
> 
> output: 输出： ![hidi_christ_27_seed_65_conditioning_scale_0 5_religious or historical artwork, Christian tradition style Alien or Xenomorph sits on a rocky outcr](https://private-user-images.githubusercontent.com/58697142/329128778-7c69c2b3-e788-422c-9a09-ba6d6c6e1977.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMzM4MTEsIm5iZiI6MTcxNTIzMzUxMSwicGF0aCI6Ii81ODY5NzE0Mi8zMjkxMjg3NzgtN2M2OWMyYjMtZTc4OC00MjJjLTlhMDktYmE2ZDZjNmUxOTc3LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDA1NDUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlNWQ3MTc5YTViNGZmMmVlMjZkNWFiNDk1OTBmYTc4MzdjMTQwZWVkODk1MmU0NmYzYjEwM2JiYTU5ZGU3NzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rKLEWtpoT5QuOZw_a82p-YV_8ZqXZ-zZq7yop5ozDOQ)

That's great! I can't wait to try it!

**Comment by Yzhbuaa** [comment]
Hello, which inpainting controlnet for SDXL did you use in the example above?

**Comment by ShenZhang-Shin** [comment]
> Hello, which inpainting controlnet for SDXL did you use in the example above?

diffusers/controlnet-canny-sdxl-1.0

**Comment by Yzhbuaa** [comment]
Thank you for your quick reply! Did you mean that you use stabilityai/stable-diffusion-xl-base-1.0 conbined with diffusers/controlnet-canny-sdxl-1.0 to perform the image inpainting task? What is the benefit of using a canny controlnet rather than only sdxl?

**Comment by ShenZhang-Shin** [comment]
> Thank you for your quick reply! Did you mean that you use stabilityai/stable-diffusion-xl-base-1.0 conbined with diffusers/controlnet-canny-sdxl-1.0 to perform the image inpainting task? What is the benefit of using a canny controlnet rather than only sdxl?

Yes.  
Only use stabilityai/stable-diffusion-xl-base-1.0 can also do inpainting task, but the pose consistency is not as good as sdxl+controlnet. 
For example, here is the inpainting output only using sdxl+hidiffusion. You can see the difference.
![9_seed_141_48 77_oil painting, Alien or Xenomorph sitting on a white rocky outcrop](https://github.com/megvii-research/HiDiffusion/assets/58697142/53186b16-2d92-49d3-a523-e279e5807ec3)
